CONTRIBUTING.md

```diff
@@ -66,6 +67,112 @@ please ensure that:
    deprecation, removal etc) that you have added a reno release note for that
    change and that the PR is tagged for the changelog.
 
+
+### Code review
+
+All code merged to Qiskit, even from maintainers, goes through a code-review
+process after a pull request is made.  There are a small number of
+maintainers who can authorize a final merge, but code review is everyone
+working together to make Qiskit better.  You can review code even if you
+are not a maintainer, which helps make sure pull requests are technically
+correct, well tested, and easier in final review.
+
+The code-review process is a normal part of software development, and nothing to
+be scared of; for very easy changes it can be as simple as a maintainer saying
+"looks good to me!" and merging the PR.  For more complex changes, it's often a
+back-and-forth where the reviewer may ask a couple of questions about why things
+were done a particular way, and make suggestions for improvement.  You don't
+need to do everything suggested if you've got good reasons to disagree, but
+communicate that clearly and politely.
+
+If you're struggling with code review or a PR on Qiskit, you can ask for help in
+the `#qiskit-pr-help` channel on [the public Qiskit Slack](https://qisk.it/join-slack).
+
+Remember that the PR author is a human, not just a screen name!  It's ok to ask
+questions about the code, but don't be mean or rude about it even if you don't
+like it.
+
+#### What to focus on in review
+
+* Is everything in [the PR checklist](#pull-request-checklist) done?
+* Are any new public APIs easy to use, well documented, and in keeping with
+  other parts of the Qiskit API?
+* Do any changes to the code have knock-on effects for other parts of Qiskit
+  that may be using them, or do the changes imply changes to the assumptions in
+  our data structures?
+* Are there any edge cases you can think of that the code might not handle well?
+* Is the code reasonably easy for you to understand?  This particular point is
+  tricky; the more you review code, the easier it will be for you to understand
+  more code, so don't worry about this as much if you're getting started.
+
+#### Writing review comments
+
+* Make concrete suggestions when you think something should be changed, but
+  remember that the author might have already thought about it and have a
+  reason.  Try "what do you think about us raising a `TypeError` here instead of
+  returning `None`?", rather than "you should raise an exception here".
+* Try to make each round of review thorough.  Don't add one or two comments on
+  one file, then come back a day later and add a couple of other unrelated
+  comments on a different file, and so on.  Try to review the whole PR
+  thoroughly in one go; it's easier to catch bugs like this, and less
+  frustrating to be on the other side of. If that's too much for you, that might
+  mean that the PR should be split into smaller chunks.
```

Comment:
```suggestion
  frustrating to be on the other side of. If that's too much for you, that might
  mean that the PR should be split into smaller chunks.
```

This would be on the PR author, right? I think we ought to drop that last sentence, or suggest that they ask the PR author to make the PR more focused.

Reply:
That'd be reasonable for them to infer. IMO though, it's a bit hand-wavy as it is, since we're putting stock on the competence of the reader to assess whether or not their review fatigue comes from a poorly factored PR.

Reply:
Fair enough - I tweaked the wording in d62c1eea2, which I think should help?